sdk/go/pulumi/run.go

```diff
@@ -38,15 +39,16 @@ type RunOption func(*RunInfo)
 // to register resources and orchestrate deployment activities.  This connects back to the Pulumi engine using gRPC.
 // If the program fails, the process will be terminated and the function will not return.
 func Run(body RunFunc, opts ...RunOption) {
-	if err := RunErr(body, opts...); err != nil {
-		if err != ErrPlugins {
-			fmt.Fprintf(os.Stderr, "error: program failed: %v\n", err)
-			os.Exit(1)
-		}
+	err := RunErr(body, opts...)
+	if err == nil {
+		return
+	}
 
+	if err == ErrPlugins {
 		printRequiredPlugins()
-		os.Exit(0)
 	}
+
+	os.Exit(constant.ExitStatusLoggedError)
```

Comment:
OK so this is the new case! Good, but sounds like something else is doing logging, it's RunERr it seems.